src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.dlic.rest.validation;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.StringJoiner;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.ImmutableList;
+import com.nulabinc.zxcvbn.Strength;
+import com.nulabinc.zxcvbn.Zxcvbn;
+import com.nulabinc.zxcvbn.matchers.Match;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.common.Strings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.dlic.rest.validation.AbstractConfigurationValidator.ErrorType;
+
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_MIN_LENGTH;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION_STRENGTH;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX;
+
+public class PasswordValidator {
+
+    private static final int MAX_LENGTH = 100;
+    private final static Predicate<Match> USERNAME_SIMILARITY_CHECK = m ->
+            m.pattern == com.nulabinc.zxcvbn.Pattern.Dictionary && m.dictionaryName.equals("user_inputs");
```

Comment:
`user_inputs` the default dictionary which library creates automatically to check similarity. I will add a comment otherwise it is confusing. So all you need is just check the dictionary name since `matchedWord` is a password.

Reply:
okay..thank yoU!